In [3]:
def patternNumber(n):
    val = 1
    
    for i in range(1,n+1):
        
        for j in range(1, i+1):
            print(val, end="")
            val = val +1
        print("\n")

In [5]:
if __name__ == "__main__":
    n = int(input())
    patternNumber(n)

 23


1

23

456

78910

1112131415

161718192021

22232425262728

2930313233343536

373839404142434445

46474849505152535455

5657585960616263646566

676869707172737475767778

79808182838485868788899091

9293949596979899100101102103104105

106107108109110111112113114115116117118119120

121122123124125126127128129130131132133134135136

137138139140141142143144145146147148149150151152153

154155156157158159160161162163164165166167168169170171

172173174175176177178179180181182183184185186187188189190

191192193194195196197198199200201202203204205206207208209210

211212213214215216217218219220221222223224225226227228229230231

232233234235236237238239240241242243244245246247248249250251252253

254255256257258259260261262263264265266267268269270271272273274275276



val = 1 - This initializes a variable val to 1. val will be used to print the numbers in the pattern.

for i in range(1, n + 1): - This for-loop iterates from 1 to n (inclusive). The variable i represents the current line number.

for j in range(1, i + 1): - This nested for-loop iterates from 1 to i (inclusive). For each line i, it runs i times. It's responsible for printing the numbers on a single line.

print(val, end="") - This prints the current value of val without moving to the next line (end="" keeps the cursor on the same line). After each print, val is incremented by 1.
print("\n") - This prints a newline character, moving the output to the next line after finishing the current line's number printing.

Main Block if __name__ == "__main__":
This is a common Python idiom for indicating the part of the script that should run when the script is executed as a standalone file (as opposed to being imported as a module in another script).

n = int(input()) - This line gets input from the user, converts it to an integer, and stores it in the variable n.

patternNumber(n) - This calls the patternNumber function with the user-provided n.

Example
If n = 3, the output would be:

Copy code
1
23
456